### PR TITLE
use Dict as storage of servs

### PR DESCRIPTION
**Pull request by Alexey-T**
dict is good to have. Now ini file is nice looking. Optons are extendable, easy to add more  keys. Namedtuple is hard to extend.

on reading old file, list of list converted to list of dict.

### CODE REVIEW

File: __init__.py [hunk 1]
```diff
@@ -302,15 +314,21 @@ def on_panel(self, ed_self, id_control, id_event):
...
-        return dlg_input_ex(
+        res = dlg_input_ex(
             3,
             "FTP server info",
-            "Address:", address,
-            "Login:", login,
-            "Password:", password,
+            "Address:", (init_server['address'] if init_server else ''),
+            "Login:", (init_server['login'] if init_server else 'anonymous'),
+            "Password:", (init_server['password'] if init_server else 'user@aol.com'),
```
Comment (pohmelie) [inline __init__.py]: This can be done better
``` python
init_server.get("password", "user@aol.com")
```

File: __init__.py [hunk 2]
```diff
@@ -79,9 +79,21 @@ def init_options(self):
...
+                    # old config, convert to dict
+                    data['servers'] = [
+                        {
+                        'address': item[0],
+                        'login': item[1],
+                        'password': item[2]
+                        } for item in d]
+                self.options = data
+                
+        #print('read op', self.options)
```
Comment (pohmelie) [inline __init__.py]: remove this?

File: __init__.py [hunk 3]
```diff
@@ -302,15 +314,21 @@ def on_panel(self, ed_self, id_control, id_event):
...
+            "Address:", (init_server['address'] if init_server else ''),
+            "Login:", (init_server['login'] if init_server else 'anonymous'),
+            "Password:", (init_server['password'] if init_server else 'user@aol.com'),
         )
+        if res is not None:
+            return {
+                'address': res[0],
+                'login': res[1],
+                'password': res[2],
+                }
```
Comment (pohmelie) [inline __init__.py]: This should be done as
``` python
return dict(zip(("address", "login", "password"), res))
```

File: __init__.py [hunk 4]
```diff
@@ -458,4 +476,5 @@ def save_options(self):
 
         with self.options_filename.open(mode="w") as fout:
 
-            json.dump(self.options, fout)
+            s = json.dumps(self.options, indent=2)
+            fout.write(s)
```
Comment (pohmelie) [inline __init__.py]: Why we need this extra `s` variable?

File: __init__.py (hunk 1 above)
Comment (Alexey-T) [inline __init__.py]: no, for None it is not ok.

File: __init__.py (hunk 2 above)
Comment (Alexey-T) [inline __init__.py]: can remove this. it is for old file.

File: __init__.py (hunk 3 above)
Comment (Alexey-T) [inline __init__.py]: Later res len will be not 3

File: __init__.py (hunk 4 above)
Comment (Alexey-T) [inline __init__.py]: no need.

File: __init__.py (hunk 1 above)
Comment (pohmelie) [inline __init__.py]: Can password be `None`?

File: __init__.py (hunk 3 above)
Comment (pohmelie) [inline __init__.py]: zip cutes by shortest one, so it does not matter how long `res` is.

File: __init__.py (hunk 1 above)
Comment (Alexey-T) [inline __init__.py]: i want functions,

- server_pw
- server_adr
- server_login

inside I can call dict.get(..., '')

File: __init__.py (hunk 3 above)
Comment (Alexey-T) [inline __init__.py]: ok. Right

**Pull request closed without merging**